sunpy/timeseries/sources/goes.py

```diff
@@ -235,8 +235,10 @@ def _parse_netcdf(filepath):
             flux_name_b = flux_name_a.replace("a", "b")
             xrsa = np.array(h5nc[flux_name_a])
             xrsb = np.array(h5nc[flux_name_b])
-            xrsa_quality = np.array(h5nc[flux_name_a.replace("flux", "flags")])
-            xrsb_quality = np.array(h5nc[flux_name_b.replace("flux", "flags")])
+            flux_flag_a = h5nc.variables.get("a_flags") or h5nc.variables.get("xrsa_flags") or h5nc.variables.get("xrsa_flag")
+            flux_flag_b = h5nc.variables.get("b_flags") or h5nc.variables.get("xrsb_flags") or h5nc.variables.get("xrsb_flag")
+            xrsa_quality = np.array(h5nc[flux_flag_a.name])
+            xrsb_quality = np.array(h5nc[flux_flag_b.name])
```

Comment:
So I encapsulated this in a walrus operator (assumed if one existed they both would) and that way it should not break again if there is a 4th name for the field.

Reply:
So, as discussed in chat, we removed the if statement here because it will just fail later when it assigns the columns to the dataframe. It is probably worth revisiting making the whole qual flag optional i.e. only assign those columns to the DF if there exist matching columns that can be parsed from the netcdf.